build/root/usr/local/bin/ci_entrypoint_nfd-operator.sh

```diff
@@ -0,0 +1,56 @@
+#! /usr/bin/env bash
+
+set -o pipefail
+set -o errexit
+set -o nounset
+
+prepare_cluster_for_nfd() {
+    toolbox/cluster/capture_environment.sh
+}
+
+validate_nfd_deployment() {
+    if ! toolbox/nfd/has_nfd_labels.sh; then
+        echo "FATAL: no NFD labels found."
+        exit 1
+    fi
+}
+
+test_master_branch() {
+    CI_IMAGE_NFD_COMMIT_CI_REPO="${1:-https://github.com/openshift/cluster-nfd-operator.git}"
+    CI_IMAGE_NFD_COMMIT_CI_REF="${2:-master}"
+
+    echo "Using Git repository ${CI_IMAGE_NFD_COMMIT_CI_REPO} with ref ${CI_IMAGE_NFD_COMMIT_CI_REF}"
+
+    CI_IMAGE_NFD_COMMIT_CI_IMAGE_TAG="ci-image"
+
+    prepare_cluster_for_nfd
+    toolbox/nfd/deploy_from_commit.sh   "${CI_IMAGE_NFD_COMMIT_CI_REPO}" \
+                                        "${CI_IMAGE_NFD_COMMIT_CI_REF}" \
+                                        "${CI_IMAGE_NFD_COMMIT_CI_IMAGE_TAG}"
+    validate_nfd_deployment
```

Comment:
I would prefer a new script `toolbox/nfd/has_nfd_labels.sh --> toolbox/nfd/wait_nfd_labels.sh`, similar to `wait/has_gpu_labels.sh`, so that you can wait&retry for "any" amount of time. Things might get slow for no reason (I'm thinking about quay.io taking too long to serve the image)

Reply:
confirmed by the test that this `sleep 30` isn't enough: 
```
$ /build/root/usr/local/bin/ci_entrypoint_nfd-operator.sh test_master_branch 
...

---
roles/nfd_has_labels/tasks/main.yml:3
TASK: nfd_has_labels : Ensure that NFD labeled some nodes
----- FAILED ----

<command> oc get nodes -lfeature.node.kubernetes.io/system-os_release.ID=rhcos

<stderr> No resources found
----- FAILED ----

# and manually a bit after
$ oc get nodes -lfeature.node.kubernetes.io/system-os_release.ID=rhcos
ip-10-0-146-212.eu-central-1.compute.internal   Ready    worker   3h21m   v1.20.0+df9c838
ip-10-0-172-7.eu-central-1.compute.internal     Ready    worker   3h21m   v1.20.0+df9c838
```

Reply:
set